external/llvm-project/mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -1046,6 +1047,27 @@ LogicalResult GPUFuncOp::verifyBody() {
   return success();
 }
 
+static LogicalResult verifyKnownLaunchSizeAttr(gpu::GPUFuncOp op,
+                                               StringRef attrName) {
+  auto maybeAttr = op->getAttr(attrName);
+  if (!maybeAttr)
+    return success();
+  auto array = maybeAttr.dyn_cast<DenseI32ArrayAttr>();
+  if (!array)
+    return op.emitOpError(attrName + " must be a dense i32 array");
+  if (array.size() != 3)
+    return op.emitOpError(attrName + " must contain exactly 3 elements");
+  return success();
```

Comment:
Add a check to make sure all sizes are positive?

Reply:
There're apparently people who'll allow `uint32_max` workgroups, at least going off the int range inference for GPU, so I don't think a positivity check is in order.